<?xml version="1.0"?>

<launch>
    <arg name="base_frame"              default="base_link"/>
    <arg name="odom_frame"              default="odom"/>
    <arg name="imu_frame"               default="imu_link"/>

    <arg name="udev_port"               default="/dev/raspbot_com_port"/>
    <arg name="baud"                    default="460800"/>

    <arg name="odom_topic"              default="wheel_odom"/>
    <arg name="imu_topic"               default="imu/data"/>
    <arg name="twist_topic"             default="cmd_vel"/>
    <arg name="speed_topic"             default="speed"/>
    <arg name="encoder_debug_topic"     default="encoder_debug"/>


    <arg name="publish_odomTF"          default="true"/>
    <arg name="publish_speed"           default="false"/>
    <arg name="publish_encoder_debug"   default="false"/>
    <arg name="publish_wheel_path"      default="false"/>

    <arg name="open_rviz"               default="false"/>



    <!-- robot model -->
    <include file="$(find raspbot_description)/launch/raspbot_description.launch" />

    <include file="$(find raspbot_bringup)/launch/raspbot_minimal.launch" >
        <arg name="base_frame"       value="$(arg base_frame)"/>
        <arg name="odom_frame"       value="$(arg odom_frame)"/>
        <arg name="imu_frame"        value="$(arg imu_frame)"/>

        <arg name="udev_port"        value="$(arg udev_port)"/>
        <arg name="baud"             value="$(arg baud)"/>

        <arg name="odom_topic"       value="$(arg odom_topic)"/>
        <arg name="imu_topic"        value="$(arg imu_topic)"/>
        <arg name="twist_topic"        value="$(arg twist_topic)"/>
        <arg name="speed_topic"        value="$(arg speed_topic)"/>
        <arg name="encoder_debug_topic"        value="$(arg encoder_debug_topic)"/>

        <arg name="publish_odomTF"   value="$(arg publish_odomTF)"/>
        <arg name="publish_speed"    value="$(arg publish_speed)"/>
        <arg name="publish_encoder_debug"   value="$(arg publish_encoder_debug)"/>
    </include>

    <!-- occupy memory and assume much time -->
    <node if="$(arg publish_wheel_path)" pkg="raspbot_bringup"  type="raspbot_bringup_visualpath_node" name="raspbot_bringup_visualpath_node" output="screen" >
        <param name="odom_frame"            type="string"        value="$(arg odom_frame)"/>
        <param name="odom_topic"            type="string"        value="$(arg odom_topic)"/>
        <param name="wheel_path_topic"      type="string"        value="$(arg wheel_path_topic)"/>
    </node>

    
    <node if="$(arg open_rviz)" pkg="rviz" type="rviz" name="rviz" args="-d  $(find raspbot_bringup)/rviz/bringup.rviz" />


</launch>
